watertap/examples/flowsheets/RO_with_energy_recovery/RO_with_energy_recovery.py

```diff
@@ -344,7 +348,12 @@ def solve(blk, solver=None, tee=False, check_termination=True):
         solver = get_solver()
     results = solver.solve(blk, tee=tee)
     if check_termination:
-        assert_optimal_termination(results)
+        if not check_optimal_termination(results):
+            _log.warning(
+                "The solver failed to converge to an optimal solution."
+                "This suggests that the user provided infeasible inputs or that the model "
+                "is poorly scaled, poorly initialized, or degenerate. "
+            )
```

Comment:
I would think in cases like these we should get rid of the `check_termination` flag. It doesn't really serve a purpose now.

Reply:
I get that the check_optimal_termination is a no-op, but isn't it still useful for the user to see the associated warning for when the solver fails. So would there be a way to display this error message w/o utilizing check/assert optimal termination?

Reply:
In this case, they wouldn't see it anyway, as the `assert_optimal_termination` would have failed first and thrown an `AssertionError`. So, it is not just a no-op, but completely redundant.

Reply:
This particular instance *isn't* a no-op though -- we take some action (sending a message to the logger) if `check_optimal_termination` returns `False`.

My comment here is we don't need the `check_termination` bool on line 350, or in this function. We should just always log the warning.

Perhaps we should write a helper function which will check the optimal termination and log a warning. It would be best if it could provide some information about where it was called though. Hmm.

Reply:
I added a simple utility function that uses the check_optimal_termination. I'm not sure what adding "some information about where it was called" would look like, but let me know if I should expand on the utility function in some way.